src/editor/build/build_manager.h

```diff
@@ -0,0 +1,45 @@
+
+#ifdef TOOLS_ENABLED
+
+#ifndef GODOT_JVM_BUILD_MANAGER_H
+#define GODOT_JVM_BUILD_MANAGER_H
+
+#include <core/error_list.h>
+#include <core/os/os.h>
+#include <core/os/thread.h>
+
+class BuildManager {
+public:
+    static BuildManager& get_instance();
+
+    bool build_project_blocking();
+
+    void build_project_non_blocking();
+
+    bool can_build_project();
+
+    bool is_build_finished();
+
+    void update_build_state();
+
+    String get_log();
```

Comment:
I suggest to return `const String&` as it is a getter, this gives indication to other devs.

Reply:
As discussed on discord: this is not possible. I need to return a copy of the string because `build_log` is written from a background thread and needs a lock to read from it.